apps/protocol-frontend/src/contexts/UserContext.tsx

```diff
@@ -36,10 +36,14 @@ interface UserContextProps {
 export const UserContextProvider: React.FC<UserContextProps> = ({
   children,
 }: UserContextProps) => {
-  const { isConnected, address, chainId, provider } = useWallet();
-  const { isAuthenticated } = useAuth();
+  const { data: signer } = useSigner();
+  const { chain } = useNetwork();
+  const { isAuthenticated, authFlow } = useAuth();
+  const { isConnected, address } = useAccount({
+    onConnect: () => authFlow(),
+    // onDisconnect:
```

Comment:
Cool, I will update this on the auth bug. Is there a function for switching wallets?